android/src/main/java/io/ably/flutter/plugin/push/ManualFlutterApplicationRunner.java

```diff
@@ -21,20 +20,27 @@
 import io.flutter.plugin.common.MethodChannel;
 import io.flutter.plugin.common.StandardMethodCodec;
 
-public class PushBackgroundIsolateRunner implements MethodChannel.MethodCallHandler {
-  private static final String TAG = PushBackgroundIsolateRunner.class.getName();
-  private static final String SHARED_PREFERENCES_KEY = "io.ably.flutter.plugin.push.PushBackgroundIsolate.SHARED_PREFERENCES_KEY";
-  private static final String BACKGROUND_MESSAGE_HANDLE_KEY = "BACKGROUND_MESSAGE_HANDLE_KEY";
+/**
+ * This class is used when the application was terminated when the push notification is received.
+ * See [PushMessagingEventHandlers.java] to see where push notifications being handled whilst the
+ * app is in the background or the foreground.
+ *
+ * This class can generified to launch the app manually for any purpose, but currently it
```

Comment:
"This class can be generalised" ....

Reply:
c4e8f56470732dffa0cf23c0830c79f9a8b52199

Reply:
Sorry to be a pain, but can you make this passive. "This class can BE generalised ...

Reply:
c06b6a3293b7690d371fb7369b06322de50809fe